students/andykwon/lesson05/mailroom_pt3.py

```diff
@@ -0,0 +1,188 @@
+
+
+"""
+simple mailroom program for a non profit
+"""
+
+import sys
+
+donors = {'Sir Isaac Newton': [100.38, 2, 4, 5000.98],
+          'Zach de la Rocha': [1000.76, 5, 235.90, 50.76],
+          'Space Ghost': [1, 5, 900000, 76.45]
+          }
+
+
+
+def thank_you():
+    """
+    Get into the thank you note generation portion of the program
+    """
+
+    while True:
+        print("\n== Send a Thank You ==\n")
+        print("1) Add a new donation (creates new donor profile if a new name is entered.\n"
+              "2) Display a list of the donors.\n"
+              "3) Back\n")
+
+        response = input(">>")
+
+        # test in put to see if it is a valid option
+        try:
+            if int(response) > 3:
+                raise ValueError
+        except ValueError:
+            print("invalid input\n")
+
+        if response == "1":
+            print("Please enter the name of the donor:\n")
+            response_name = input(">>")
+            print("Please enter the amount " + response_name + " donated:\n")
+            response_donation = input(">>")
+
+            # test to see if the value inputted is a number (float)
+            try:
+                response_donation = float(response_donation)
+            except ValueError:
+                print("That was not a valid input.\n")
+                continue
+
+            # comp_names = [name.upper() for name in donors]
+            # print(comp_names)
+
+            if response_name.upper() in donors:
```

Comment:
Found a bug here because I added another donation for "Zach de la Rocha" because "ZACH DE LA ROCHA" is not in donors. So yeah, the comp_names that you have commented out above is what you should compare response_name.upper() to. You already had the answer!

Or it'd be more efficient if you should stored all names in the dict initially as .title() and you can compare response_name.title() to donors directly.